lib/smart_proxy_openscap/openscap_report_parser.rb

```diff
@@ -0,0 +1,85 @@
+# encoding=utf-8
+require 'openscap'
+require 'openscap/ds/arf'
+require 'openscap/xccdf/testresult'
+require 'openscap/xccdf/ruleresult'
+require 'openscap/xccdf/rule'
+require 'openscap/xccdf/fix'
+require 'openscap/xccdf/benchmark'
+require 'json'
+module Proxy::OpenSCAP
+  class Parse
+    def initialize(arf_data)
+      OpenSCAP.oscap_init
+      size         = arf_data.size
+      @arf_digest  = Digest::SHA256.hexdigest(arf_data)
+      @arf         = OpenSCAP::DS::Arf.new(:content => arf_data, :path => 'arf.xml.bz2', :length => size)
+      @results     = @arf.test_result.rr
+      sds          = @arf.report_request
+      bench_source = sds.select_checklist!
+      @bench       = OpenSCAP::Xccdf::Benchmark.new(bench_source)
+      @items       = @bench.items
+    end
+
+    def as_json
+      parse_report.to_json
+    end
+
+    private
+
+    def parse_report
+      report        = {}
+      report[:logs] = []
+      passed        = 0
+      failed        = 0
+      other         = 0
+      @results.each do |rr_id, result|
+        next if result.result == 'notapplicable' || result.result == 'notselected'
+        # get rules and their results
+        rule_data = @items[rr_id]
+        report[:logs] << populate_result_data(rr_id, result.result, rule_data)
+        # create metrics for the results
+        case result.result
+          when 'pass', 'fixed'
+            passed += 1
+          when 'fail'
+            failed += 1
+          else
+            other += 1
+        end
+      end
+      report[:digest]  = @arf_digest
+      report[:metrics] = { :passed => passed, :failed => failed, :other => other }
+      report
+    end
+
+    def populate_result_data(result_id, rule_result, rule_data)
+      log               = {}
+      log[:source]      = a8b(result_id)
+      log[:result]      = a8b(rule_result)
+      log[:title]       = a8b(rule_data.title)
+      log[:description] = a8b(rule_data.description)
+      log[:rationale]   = a8b(rule_data.rationale)
+      log[:references]  = hash_a8b(rule_data.references.map(&:to_hash))
+      log[:fixes]       = hash_a8b(rule_data.fixes.map(&:to_hash))
+      log[:severity]    = a8b(rule_data.severity)
+      log
+    end
+
+    def a8b(string)
```

Comment:
Could you either give it more descriptive name or add some comment above? I'm not sure what's difference between :invalid and :undef.